src/components/app-scope.js

```diff
@@ -14,41 +46,200 @@ class AppScope extends React.Component {
     }
     return dict
   }
+  convertAlreadySelectedDictionary = options => {
+    let dict = {}
+    for (let i = 0; i < options.length; i++) {
+      if(!dict[options[i]]){
+        dict[options[i]]=true;
+      }
+    }
+    return dict
+  }
+  convertOptionsListDictionary = options => {
```

Comment:
`convertDictionary` can be used here instead of creating the new function `convertOptionsListDictionary` by transformaing the `options` parameter.
Both function are nearly identical.

Reply:
To do the work done by `convertOptionsListDictionary`, I need to make a new array in the function itself which will hold only the value keys of the optionsList data getting from redux, and then pass that array into `convertDictionary` function. Without making a new array, it isn't possible ig. So how should I proceed.

Reply:
Don't make the new array inside the function. First make a new array and then pass it to function.

Reply:
It's better to keep the two functions separate instead of using one in this case because the parameter will require an array different from the choices array. This array would not be an array of objects, but of values, so its not possible to get the displayName attribute, and understanding code could become complex.